src/rhsm/connection.py

```diff
@@ -787,7 +801,16 @@ def _create_connection(self, cert_file: str = None, key_file: str = None) -> htt
         conn.max_requests_num = None
 
         # Do TCP and TLS handshake here before we make any request
-        conn.connect()
+        try:
+            conn.connect()
+        except OSError as e:
+            # in case this OSError does not have an errno set, it means it was
+            # not a syscall failure; mostly (if at all) this is raisen on proxy
+            # connection failures
+            if e.errno is None:
+                # wrap this to carry also the details on the destination host
+                raise ConnectionOSErrorException(self.host, self.ssl_port, self.apihandler, e)
+            raise
```

Comment:
In other cases we could raise similar exception containing `self.host`, `self.ssl_port`, `self.apihandler` and print this information to console too. But it should be done as part of another PR.